tests/unit/amo/components/TestCollectionAddAddon.js

```diff
@@ -0,0 +1,262 @@
+import * as React from 'react';
+/* global window */
+
+import AutoSearchInput from 'amo/components/AutoSearchInput';
+import CollectionAddAddon, {
+  CollectionAddAddonBase,
+  extractId,
+  mapStateToProps,
+  MESSAGE_RESET_TIME,
+} from 'amo/components/CollectionAddAddon';
+import {
+  addAddonToCollection,
+  addonAddedToCollection,
+  createInternalCollection,
+} from 'amo/reducers/collections';
+import { createInternalSuggestion } from 'core/reducers/autocomplete';
+import {
+  applyUIStateChanges,
+  createStubErrorHandler,
+  fakeI18n,
+  shallowUntilTarget,
+  simulateComponentCallback,
+} from 'tests/unit/helpers';
+import {
+  createFakeAutocompleteResult,
+  createFakeCollectionDetail,
+  dispatchClientMetadata,
+  dispatchSignInActions,
+} from 'tests/unit/amo/helpers';
+import ErrorList from 'ui/components/ErrorList';
+import Notice from 'ui/components/Notice';
+
+const simulateAutoSearchCallback = (props = {}) => {
+  return simulateComponentCallback({
+    Component: AutoSearchInput,
+    ...props,
+  });
+};
+
+const setAddonWasAddedUiStateToTrue = ({ username, root, store }) => {
```

Comment:
I think a better name for this would be `addAddonToCollection` or something that conveys the user's *action* rather than the component's internal representation of the user's action.

We typically name these types of functions the same as the dispatched action but with an underscore prefix. In this case: `_addonAddedToCollection`. It's not a hard and fast rule, though.